packages/evolution-frontend/src/components/survey/Button.tsx

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2024, Polytechnique Montreal and contributors
+ *
+ * This file is licensed under the MIT License.
+ * License text available at https://opensource.org/licenses/MIT
+ */
+import React, { useState, useEffect } from 'react';
+import { withTranslation, WithTranslation } from 'react-i18next';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { withSurveyContext, WithSurveyContextProps } from '../hoc/WithSurveyContextHoc';
+import ConfirmModal from 'chaire-lib-frontend/lib/components/modal/ConfirmModal';
+import { ButtonWidgetConfig } from 'evolution-common/lib/services/widgets';
+import { WidgetStatus } from '../../services/interviews/interview';
+import * as surveyHelper from 'evolution-common/lib/utils/helpers';
+import { UserInterviewAttributes } from 'evolution-common/lib/services/interviews/interview';
+import { CliUser } from 'chaire-lib-common/lib/services/user/userType';
+
+type ButtonProps = surveyHelper.InterviewUpdateCallbacks & {
+    widgetConfig: ButtonWidgetConfig;
+    widgetStatus: WidgetStatus;
+    loadingState: number;
+    // FIXME: Why is the confirmModal controlled by the parent and not this
```

Comment:
@kaligrafy See this FIXME any reason why modal management needs to go all the way up to the Survey object instead of being controlled by the widget that actually opens it?